armi/reactor/blueprints/__init__.py

```diff
@@ -393,69 +394,77 @@ def _resolveNuclides(self, cs):
         for nucFlag in self.nuclideFlags:
             # this returns any nuclides that are flagged specifically for expansion by input
-            expandedElements = nucFlag.fileAsActiveOrInert(
-                actives, inerts, undefBurnChainActiveNuclides
+            (
+                expandedElements,
+                undefBurnChainActiveNuclides,
+            ) = nucFlag.fileAsActiveOrInert(
+                actives,
+                inerts,
             )
             self.elementsToExpand.extend(expandedElements)
 
         inerts -= actives
         self.customIsotopics = self.customIsotopics or isotopicOptions.CustomIsotopics()
-        (
-            elementalsToKeep,
-            expansions,
-        ) = isotopicOptions.autoSelectElementsToKeepFromSettings(cs)
+        eleKeep, eleExpand = isotopicOptions.eleExpandInfoBasedOnCodeENDF(cs)
 
         # Flag all elementals for expansion unless they've been flagged otherwise by
         # user input or automatic lattice/datalib rules.
-        for elemental in nuclideBases.instances:
-            if not isinstance(elemental, nuclideBases.NaturalNuclideBase):
+        for nucBase in nuclideBases.instances:
+            wasInOriginalFlags = nucBase.name in originalFlagKeys
+            isAlreadyIsotopic = not isinstance(nucBase, nuclideBases.NaturalNuclideBase)
+            if isAlreadyIsotopic:
+                if wasInOriginalFlags:
+                    nucsToForceInXsGen.add(nucBase.name)
                 # `elemental` may be a NaturalNuclideBase or a NuclideBase
-                # skip all NuclideBases
+                # skip all NuclideBases (isotopics)
                 continue
 
-            if elemental in elementalsToKeep:
+            # we now know its an elemental
+            elemental = nucBase
+            if elemental in eleKeep:
+                if wasInOriginalFlags:
+                    nucsToForceInXsGen.add(elemental.name)
                 continue
-
             if elemental.name in actives:
                 currentSet = actives
-                actives.remove(elemental.name)
             elif elemental.name in inerts:
                 currentSet = inerts
-                inerts.remove(elemental.name)
             else:
                 # This was not specified in the nuclide flags at all.
                 # If a material with this in its composition is brought in
                 # it's nice from a user perspective to allow it.
                 # But current behavior is that all nuclides in problem
                 # must be declared up front.
+                if wasInOriginalFlags:
+                    nucsToForceInXsGen.add(elemental.name)
                 continue
 
             self.elementsToExpand.append(elemental.element)
 
             if (
                 elemental.name in self.nuclideFlags
-                and self.nuclideFlags[elemental.name].expandTo
+                and self.nuclideFlags[elemental.name.symbol].expandTo
```

Comment:
we know its an elemental already, and for elemental, name and symbol are same. rest of this uses symbol so symbol for consistency. otherwise the  developer has no  idea why its different here. (i double checked name== symbol for all ele)

Reply:
I'm confused by `elemental.name.symbol` vs. `elemental.element.symbol`

Could the usage in lines 456, 461, and 464 all be unified to the same attribute?

Reply:
yes should be `elemental.element.symbol`